# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/atk/nsMaiInterfaceTableCell.cpp
# Commit: ba629ad14280
# Full Hash: ba629ad14280ff2182a3a1f16d7ed86982a0a914
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-10-19 04:09:55
# Description:
#   Bug 1794719: Fix retrieval of AtkObject from cached RemoteAccessible in GetColumn/RowHeaderCellsCB. r=eeejay
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159238
# ==============================================================================

diff -r 6eb4e2bde9c3 -r ba629ad14280 accessible/atk/nsMaiInterfaceTableCell.cpp
--- a/accessible/atk/nsMaiInterfaceTableCell.cpp	Tue Oct 18 21:57:53 2022 +0000
+++ b/accessible/atk/nsMaiInterfaceTableCell.cpp	Tue Oct 18 21:58:34 2022 +0000
@@ -145,7 +145,7 @@
 
     GPtrArray* atkHeaders = g_ptr_array_sized_new(headers.Length());
     for (Accessible* header : headers) {
-      AtkObject* atkHeader = AccessibleWrap::GetAtkObject(header->AsLocal());
+      AtkObject* atkHeader = GetWrapperFor(header);
       g_object_ref(atkHeader);
       g_ptr_array_add(atkHeaders, atkHeader);
     }
@@ -187,7 +187,7 @@
 
     GPtrArray* atkHeaders = g_ptr_array_sized_new(headers.Length());
     for (Accessible* header : headers) {
-      AtkObject* atkHeader = AccessibleWrap::GetAtkObject(header->AsLocal());
+      AtkObject* atkHeader = GetWrapperFor(header);
       g_object_ref(atkHeader);
       g_ptr_array_add(atkHeaders, atkHeader);
     }
